internal/datastore/postgres/readwrite.go

```diff
@@ -0,0 +1,258 @@
+			bulkWrite = bulkWrite.Values(
+				rel.Resource.ObjectType,
+				rel.Resource.ObjectId,
+				rel.Relation,
+				rel.Subject.Object.ObjectType,
+				rel.Subject.Object.ObjectId,
+				stringz.DefaultEmpty(rel.Subject.OptionalRelation, datastore.Ellipsis),
+				rwt.newTxnID,
+			)
+			bulkWriteHasValues = true
+		}
+	}
+
+	if bulkWriteHasValues {
+		sql, args, err := bulkWrite.ToSql()
+		if err != nil {
+			return fmt.Errorf(errUnableToWriteRelationships, err)
+		}
+
+		_, err = rwt.tx.Exec(ctx, sql, args...)
+		if err != nil {
+			return fmt.Errorf(errUnableToWriteRelationships, err)
+		}
+	}
+
+	return nil
+}
+
+func (rwt *pgReadWriteTXN) DeleteRelationships(filter *v1.RelationshipFilter) error {
+	ctx, span := tracer.Start(datastore.SeparateContextWithTracing(rwt.ctx), "DeleteRelationships")
+	defer span.End()
+
+	// Add clauses for the ResourceFilter
+	query := deleteTuple.Where(sq.Eq{colNamespace: filter.ResourceType})
+	tracerAttributes := []attribute.KeyValue{common.ObjNamespaceNameKey.String(filter.ResourceType)}
+	if filter.OptionalResourceId != "" {
+		query = query.Where(sq.Eq{colObjectID: filter.OptionalResourceId})
+		tracerAttributes = append(tracerAttributes, common.ObjIDKey.String(filter.OptionalResourceId))
+	}
+	if filter.OptionalRelation != "" {
+		query = query.Where(sq.Eq{colRelation: filter.OptionalRelation})
+		tracerAttributes = append(tracerAttributes, common.ObjRelationNameKey.String(filter.OptionalRelation))
+	}
+
+	// Add clauses for the SubjectFilter
+	if subjectFilter := filter.OptionalSubjectFilter; subjectFilter != nil {
+		query = query.Where(sq.Eq{colUsersetNamespace: subjectFilter.SubjectType})
+		tracerAttributes = append(tracerAttributes, common.SubNamespaceNameKey.String(subjectFilter.SubjectType))
+		if subjectFilter.OptionalSubjectId != "" {
+			query = query.Where(sq.Eq{colUsersetObjectID: subjectFilter.OptionalSubjectId})
+			tracerAttributes = append(tracerAttributes, common.SubObjectIDKey.String(subjectFilter.OptionalSubjectId))
+		}
+		if relationFilter := subjectFilter.OptionalRelation; relationFilter != nil {
+			query = query.Where(sq.Eq{colUsersetRelation: stringz.DefaultEmpty(relationFilter.Relation, datastore.Ellipsis)})
+			tracerAttributes = append(tracerAttributes, common.SubRelationNameKey.String(relationFilter.Relation))
+		}
+	}
+
+	span.SetAttributes(tracerAttributes...)
+
+	query = query.Set(colDeletedTxn, rwt.newTxnID)
+
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	if _, err := rwt.tx.Exec(ctx, sql, args...); err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	return nil
+}
+
+func (rwt *pgReadWriteTXN) WriteNamespaces(newConfigs ...*core.NamespaceDefinition) error {
+	ctx := datastore.SeparateContextWithTracing(rwt.ctx)
+
+	ctx, span := tracer.Start(ctx, "WriteNamespaces")
+	defer span.End()
+
+	// span.SetAttributes(common.ObjNamespaceNameKey.String(newConfig.Name))
```

Comment:
delete this or leave a comment describing it

Reply:
Just like with MySQL, I don't know what to set it to now that the method is vectorized. Suggestions?